scheds/rust/scx_layered/src/main.rs

```diff
@@ -2147,6 +2168,239 @@ impl<'a> Scheduler<'a> {
         Ok(())
     }
 
+    fn calc_target_llcs(&self) -> Vec<usize> {
+        let nr_llcs = self.cpu_pool.topo.all_llcs.len();
+        let mut wants = vec![0; self.layers.len()];
+
+        for (i, layer) in self.layers.iter().enumerate() {
+            let owned = self.sched_stats.layer_utils[i][LAYER_USAGE_OWNED];
+            let open = self.sched_stats.layer_utils[i][LAYER_USAGE_OPEN];
+            let total_util = owned + open;
+
+            // "raw" guess ignoring hysteresis
+            let raw = (total_util * nr_llcs as f64).ceil() as usize;
+            // Always have one LLC
+            let raw_clamped = raw.clamp(1, nr_llcs);
+
+            // current assigned
+            let curr = layer.llcs_assigned.len().max(1);
+
+            // If we've been above threshold enough intervals => grow
+            if layer.above_count >= layer.intervals_required {
+                wants[i] = (curr + 1).max(raw_clamped);
+            }
+            // If we've been below threshold => shrink
+            else if layer.below_count >= layer.intervals_required {
+                let new_count = if curr > 1 { curr - 1 } else { curr };
+                wants[i] = new_count.min(raw_clamped);
+            }
+            // Otherwise, we stay as-is
+            else {
+                wants[i] = curr;
+            }
+
+            if layer.is_sticky && wants[i] < curr {
+                wants[i] = curr;
+            }
+        }
+        wants
+    }
+
+    fn weighted_target_llcs(&self, raw_wants: &[usize]) -> Vec<usize> {
```

Comment:
Ditto, wouldn't it make more sense to determine this according to the number of CPUs allocated to each layer?